File: RiskProfileService.py
class RiskProfileService:
    def __init__(self):
        self.scorer = RiskScorer()
        self.allocator = PortfolioAllocator()

    def generate_profile(self, form: dict) -> dict:
        score = self.scorer.calculate_score(form)
        risk_band = self.scorer.classify(score)
        portfolio = self.allocator.get_allocation(risk_band)

        return {
            "risk_score": score,
            "risk_band": risk_band,
            "portfolio_allocation": portfolio
        }

File: backend/main_server_utils/utils/risk_profile_service.js
const RiskScorer = require('./risk_scorer.js');
const PortfolioAllocator = require('./portfolio_allocator.js');

class RiskProfileService {
  constructor() {
    this.scorer = new RiskScorer();
    this.allocator = new PortfolioAllocator();
  }

  generateProfile(form) {
    const score = this.scorer.calculateScore(form);
    const riskBand = this.scorer.classify(score);
    const portfolio = this.allocator.getAllocation(riskBand);

    return {
      risk_score: score,
      risk_band: riskBand,
      portfolio_allocation: portfolio
    };
  }
}

module.exports = RiskProfileService;